### ISSUE TITLE
Replace HTML parser

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#3817]
Need to investigate if we can replace our fragile existing HTML tokenizer/parser with something like DocumentFragment.

Advantages:
- Replace our custom, fragile code with a supported codebase.
- Use browser's native features, which should be faster for queries, etc.
- Better support for newer HTML tags/constructs.

To investigate:
- [DocumentFragment](https://developer.mozilla.org/en-US/docs/DOM/DocumentFragment)
- [DOMParser](https://developer.mozilla.org/en-US/docs/DOM/DOMParser)
- Do these model the full HTML document (including `<head>`)?
- Is JavaScript (inline and from script tags) run when modeled, or only when added to a "real" DOM?
- How is CSS handled?  Is it parsed as well?
- How are errors handled?  Is there decent error recoverability?
- [node-htmlparser](https://github.com/tautologistics/node-htmlparser)
  - [htmlparser2](https://github.com/fb55/node-htmlparser) is newer (original was last updated 11 months ago), but only runs in NodeJS.

@peller @JonFerraiolo @billreed63

**Comment by peller** [comment]
I think the biggest problems are correlating this DOM back to the original source.  Although you can generate source from a DOM, it doesn't necessarily have the same line/number positions and may in fact have a slightly different structure, right?

**Comment by jhpedemonte** [comment]
That's true.  Right now, out Model keeps track of line numbers and correlates selection in the source to selection in the VE.

We could probably add such metadata as properties to the DOM nodes, or as a structure on top, but that may not be easy.

**Comment by peller** [comment]
but where would we get the line/column metadata without a non-native parser?
